src/app/wallets/update-pending-payments.ts

```diff
@@ -59,21 +61,45 @@ const updatePendingPayment = async ({
     throw new InconsistentDataError("paymentHash missing from payment transaction")
   if (!pubkey) throw new InconsistentDataError("pubkey missing from payment transaction")
 
-  const lnPaymentLookup = await lndService.lookupPayment({
+  const lnPaymentLookupFromLightningPromise = lndService.lookupPayment({
     pubkey,
     paymentHash,
   })
+  const lnPaymentLookupPromise = LnPaymentsRepository().findByPaymentHash(paymentHash)
+
+  const lightningLogger = logger.child({
+    topic: "payment",
+    protocol: "lightning",
+    transactionType: "payment",
+    onUs: false,
+  })
+
+  const [lnPaymentLookupFromLightning, lnPaymentLookup] = await Promise.all([
+    lnPaymentLookupFromLightningPromise,
+    lnPaymentLookupPromise,
+  ])
+  if (lnPaymentLookupFromLightning instanceof Error) {
+    lightningLogger.error(
+      { err: lnPaymentLookupFromLightning },
+      "issue fetching payment from lightning service",
+    )
+    return lnPaymentLookupFromLightning
+  }
   if (lnPaymentLookup instanceof Error) {
-    const lightningLogger = logger.child({
-      topic: "payment",
-      protocol: "lightning",
-      transactionType: "payment",
-      onUs: false,
-    })
-    lightningLogger.error({ err: lnPaymentLookup }, "issue fetching payment")
+    lightningLogger.error(
+      { err: lnPaymentLookup },
+      "issue fetching payment from database",
+    )
     return lnPaymentLookup
```

Comment:
I'm not sure we should fail if there is an issue with `lnPaymentLookupPromise`, ie: `LnPaymentsRepository().findByPaymentHash(paymentHash)`. 

this is a non mandatory metadata, so we could just log the error and continue?

Reply:
Sure that makes sense. We could optionally always retrieve this later on as well from lnd.

Reply:
Done https://github.com/GaloyMoney/galoy/pull/665/commits/b79ea2f740c6169dde892407e1e079b32a0614b5